cdc/verification/checker.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verification
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/log"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/filter"
+	"go.uber.org/zap"
+)
+
+//go:generate mockery --name=checkSumChecker --inpackage
+type checkSumChecker interface {
+	getCheckSum(ctx context.Context, db string, f *filter.Filter) (map[string]string, error)
+	getAllDBs(ctx context.Context) ([]string, error)
+}
+
+type checker struct {
+	db *sql.DB
+}
+
+// nolint:deadcode, will be used in next pr
+func newChecker(db *sql.DB) *checker {
+	return &checker{
+		db: db,
+	}
+}
+
+func (c *checker) getCheckSum(ctx context.Context, db string, f *filter.Filter) (map[string]string, error) {
+	_, err := c.db.ExecContext(ctx, fmt.Sprintf("USE %s", db))
+	if err != nil {
+		return nil, cerror.WrapError(cerror.ErrMySQLQueryError, err)
+	}
+
+	tables, err := c.getAllTables(ctx, db, f)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[string]string)
+	for _, table := range tables {
+		columns, err := c.getColumns(ctx, table)
+		if err != nil {
+			return result, err
+		}
+		checkSum, err := c.doChecksum(ctx, columns, db, table)
+		if err != nil {
+			return result, err
+		}
+		result[table] = checkSum
+	}
+
+	return result, nil
+}
+
+func (c *checker) getAllDBs(ctx context.Context) ([]string, error) {
```

Comment:
it looks that `getAllDBs` and `getAllTables` are almost identical, except for the SQL and some filtering logic.

How about abstracting another function to do the data fetch work, and then do specific logic in this 2 methods.

Reply:
i do not get your point